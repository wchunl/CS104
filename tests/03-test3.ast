ROOT "" (0.0.0)
|   TOK_FUNCTION "" (6.22.1) function
|   |   TOK_TYPE_ID "" (6.22.1) typeid
|   |   |   TOK_VOID "void" (6.22.1) void
|   |   |   TOK_IDENT "fail" (6.22.6) variable
|   |   TOK_PARAM "(" (6.22.11) param
|   |   |   TOK_TYPE_ID "" (6.22.12) typeid
|   |   |   |   TOK_STRING "string" (6.22.12) string
|   |   |   |   TOK_IDENT "expr" (6.22.19) string variable (6.22.12)
|   |   |   TOK_TYPE_ID "" (6.22.25) typeid
|   |   |   |   TOK_STRING "string" (6.22.25) string
|   |   |   |   TOK_IDENT "file" (6.22.32) string variable (6.22.25)
|   |   |   TOK_TYPE_ID "" (6.22.38) typeid
|   |   |   |   TOK_INT "int" (6.22.38) int
|   |   |   |   TOK_IDENT "line" (6.22.42) int variable (6.22.38)
|   TOK_FUNCTION "" (6.24.1) function
|   |   TOK_TYPE_ID "" (6.24.1) typeid
|   |   |   TOK_VOID "void" (6.24.1) void
|   |   |   TOK_IDENT "putchr" (6.24.6) variable
|   |   TOK_PARAM "(" (6.24.13) param
|   |   |   TOK_TYPE_ID "" (6.24.14) typeid
|   |   |   |   TOK_INT "int" (6.24.14) int
|   |   |   |   TOK_IDENT "chr" (6.24.18) int variable (6.24.14)
|   TOK_FUNCTION "" (6.25.1) function
|   |   TOK_TYPE_ID "" (6.25.1) typeid
|   |   |   TOK_VOID "void" (6.25.1) void
|   |   |   TOK_IDENT "putint" (6.25.6) variable
|   |   TOK_PARAM "(" (6.25.13) param
|   |   |   TOK_TYPE_ID "" (6.25.14) typeid
|   |   |   |   TOK_INT "int" (6.25.14) int
|   |   |   |   TOK_IDENT "num" (6.25.18) int variable (6.25.14)
|   TOK_FUNCTION "" (6.26.1) function
|   |   TOK_TYPE_ID "" (6.26.1) typeid
|   |   |   TOK_VOID "void" (6.26.1) void
|   |   |   TOK_IDENT "putstr" (6.26.6) variable
|   |   TOK_PARAM "(" (6.26.13) param
|   |   |   TOK_TYPE_ID "" (6.26.14) typeid
|   |   |   |   TOK_STRING "string" (6.26.14) string
|   |   |   |   TOK_IDENT "str" (6.26.21) string variable (6.26.14)
|   TOK_FUNCTION "" (6.28.1) function
|   |   TOK_TYPE_ID "" (6.28.1) typeid
|   |   |   TOK_INT "int" (6.28.1) int
|   |   |   TOK_IDENT "getchr" (6.28.5) variable
|   |   TOK_PARAM "(" (6.28.11) param
|   TOK_FUNCTION "" (6.29.1) function
|   |   TOK_TYPE_ID "" (6.29.1) typeid
|   |   |   TOK_STRING "string" (6.29.1) string
|   |   |   TOK_IDENT "getstr" (6.29.8) variable
|   |   TOK_PARAM "(" (6.29.14) param
|   TOK_FUNCTION "" (6.30.1) function
|   |   TOK_TYPE_ID "" (6.30.1) typeid
|   |   |   TOK_STRING "string" (6.30.1) string
|   |   |   TOK_IDENT "getln" (6.30.8) variable
|   |   TOK_PARAM "(" (6.30.13) param
|   TOK_FUNCTION "" (7.5.1) function
|   |   TOK_TYPE_ID "" (7.5.1) typeid
|   |   |   TOK_INT "int" (7.5.1) int
|   |   |   TOK_IDENT "main" (7.5.5) variable
|   |   TOK_PARAM "(" (7.5.9) param
|   |   TOK_BLOCK "{" (7.5.12)
|   |   |   TOK_VARDECL "=" (7.6.10) variable lval
|   |   |   |   TOK_INT "int" (7.6.4) int
|   |   |   |   TOK_IDENT "a" (7.6.8) int variable (7.6.10)
|   |   |   |   TOK_INTCON "3" (7.6.12) int const
|   |   |   TOK_VARDECL "=" (7.7.10) variable lval
|   |   |   |   TOK_INT "int" (7.7.4) int
|   |   |   |   TOK_IDENT "b" (7.7.8) int variable (7.7.10)
|   |   |   |   TOK_INTCON "8" (7.7.12) int const
|   |   |   TOK_VARDECL "=" (7.8.10) variable lval
|   |   |   |   TOK_INT "int" (7.8.4) int
|   |   |   |   TOK_IDENT "c" (7.8.8) int variable (7.8.10)
|   |   |   |   '+' "+" (7.8.14) int vreg
|   |   |   |   |   TOK_IDENT "a" (7.8.12) int variable (7.6.10)
|   |   |   |   |   TOK_IDENT "b" (7.8.16) int variable (7.7.10)
|   |   |   '=' "=" (7.9.6) int vreg
|   |   |   |   TOK_IDENT "a" (7.9.4) int variable (7.6.10)
|   |   |   |   '+' "+" (7.9.10) int vreg
|   |   |   |   |   TOK_IDENT "b" (7.9.8) int variable (7.7.10)
|   |   |   |   |   TOK_IDENT "c" (7.9.12) int variable (7.8.10)
|   |   |   TOK_CALL "(" (7.10.11)
|   |   |   |   TOK_IDENT "putint" (7.10.4) variable
|   |   |   |   TOK_IDENT "a" (7.10.12) int variable (7.6.10)
|   |   |   TOK_CALL "(" (7.11.11)
|   |   |   |   TOK_IDENT "putchr" (7.11.4) variable
|   |   |   |   TOK_CHARCON "'\n'" (7.11.12) int const
|   |   |   TOK_RETURN "return" (7.12.4)
|   |   |   |   TOK_INTCON "0" (7.12.11) int const
