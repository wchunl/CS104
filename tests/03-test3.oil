fail:     .function
          .param ptr expr
          .param ptr file
          .param int line
          .end
putchr:   .function
          .param int chr
          .end
putint:   .function
          .param int num
          .end
putstr:   .function
          .param ptr str
          .end
getchr:   .function int
          .end
getstr:   .function
          .end
getln:    .function
          .end
main:     .function int
          .local int a
          .local int b
          .local int c
          a = 3
          b = 8
          $t0:i = a + b
          c = $t0:i
          $t1:i = b + c
          a = $t1:i
          putint(a)
          putchr('\n')
          return 0
          return 
          .end
