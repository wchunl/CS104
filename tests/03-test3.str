string_set[   0]:    4706206694200543916 0xabadc8->","
                   16579916199511856337 0xabc488->"getln"
string_set[   1]:    2297668033614959926 0xabd928->"0"
string_set[   3]:    8248777770799913213 0xabccd8->"3"
                   11868120863286193964 0xabc6f8->"main"
                    9611428142580219387 0xab6758->"fail"
string_set[   4]:   10838281452030117757 0xabcd88->"b"
                      21606623943142546 0xabc998->"{"
                    4017852018407011111 0xabaed8->"file"
string_set[   6]:   13686721134187162920 0xabd9f8->"}"
string_set[   8]:    1370730858159036685 0xabcff8->"+"
                    9386170080689500517 0xabac78->"expr"
string_set[  12]:   15816506932648033945 0xabaae8->"("
string_set[  17]:   17899137375351314596 0xabcf48->"8"
                   18413295556186398326 0xabb2c8->"putint"
string_set[  18]:    7211130878751977441 0xabb418->"putchr"
                   12679825168387489523 0xabb9a8->"num"
string_set[  19]:   10959529184379665549 0xabcda8->"c"
string_set[  23]:   11166701219533395393 0xabc1f8->"getstr"
                   12868445110721718657 0xabaef8->")"
string_set[  25]:    4993892634952068459 0xabb268->"a"
string_set[  28]:   14635384904416252381 0xabd858->"return"
string_set[  29]:   12291392672316406109 0xabb0b8->"line"
string_set[  31]:    6192799629375248351 0xabd018->"'\n'"
                    6142509188972423790 0xab6738->""
                   13348923631108115385 0xabb738->"putstr"
string_set[  32]:   11597697714117577063 0xabcc28->"="
string_set[  33]:    1583576014053361350 0xabbb58->"str"
string_set[  38]:   12597418120160561823 0xabacf8->"int"
string_set[  39]:    7296690267450279874 0xabb588->"chr"
string_set[  40]:   11911796782241914524 0xab6318->"void"
string_set[  42]:    5344662657899890615 0xabb0d8->";"
string_set[  44]:   13497578857207247412 0xabab98->"string"
string_set[  46]:    9201800968376702500 0xabbf88->"getchr"
load_factor = 0.723
bucket_count = 47
max_bucket_size = 3
