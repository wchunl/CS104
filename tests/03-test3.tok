#  1 "03-test3.oc"
#  1 "<built-in>"
#  1 "<command-line>"
#  1 "03-test3.oc"
#  1 "oclib.h"
# 22 "oclib.h"
   6 22.00001 261 TOK_VOID      void
   6 22.00006 282 TOK_IDENT     fail
   6 22.00011  40 '('           (
   6 22.00012 263 TOK_STRING    string
   6 22.00019 282 TOK_IDENT     expr
   6 22.00023  44 ','           ,
   6 22.00025 263 TOK_STRING    string
   6 22.00032 282 TOK_IDENT     file
   6 22.00036  44 ','           ,
   6 22.00038 262 TOK_INT       int
   6 22.00042 282 TOK_IDENT     line
   6 22.00046  41 ')'           )
   6 22.00047  59 ';'           ;
   6 24.00001 261 TOK_VOID      void
   6 24.00006 282 TOK_IDENT     putchr
   6 24.00013  40 '('           (
   6 24.00014 262 TOK_INT       int
   6 24.00018 282 TOK_IDENT     chr
   6 24.00021  41 ')'           )
   6 24.00022  59 ';'           ;
   6 25.00001 261 TOK_VOID      void
   6 25.00006 282 TOK_IDENT     putint
   6 25.00013  40 '('           (
   6 25.00014 262 TOK_INT       int
   6 25.00018 282 TOK_IDENT     num
   6 25.00021  41 ')'           )
   6 25.00022  59 ';'           ;
   6 26.00001 261 TOK_VOID      void
   6 26.00006 282 TOK_IDENT     putstr
   6 26.00013  40 '('           (
   6 26.00014 263 TOK_STRING    string
   6 26.00021 282 TOK_IDENT     str
   6 26.00024  41 ')'           )
   6 26.00025  59 ';'           ;
   6 28.00001 262 TOK_INT       int
   6 28.00005 282 TOK_IDENT     getchr
   6 28.00011  40 '('           (
   6 28.00012  41 ')'           )
   6 28.00013  59 ';'           ;
   6 29.00001 263 TOK_STRING    string
   6 29.00008 282 TOK_IDENT     getstr
   6 29.00014  40 '('           (
   6 29.00015  41 ')'           )
   6 29.00016  59 ';'           ;
   6 30.00001 263 TOK_STRING    string
   6 30.00008 282 TOK_IDENT     getln
   6 30.00013  40 '('           (
   6 30.00014  41 ')'           )
   6 30.00015  59 ';'           ;
#  4 "03-test3.oc"
   7 5.00001 262 TOK_INT       int
   7 5.00005 282 TOK_IDENT     main
   7 5.00009  40 '('           (
   7 5.00010  41 ')'           )
   7 5.00012 123 '{'           {
   7 6.00004 262 TOK_INT       int
   7 6.00008 282 TOK_IDENT     a
   7 6.00010  61 '='           =
   7 6.00012 283 TOK_INTCON    3
   7 6.00013  59 ';'           ;
   7 7.00004 262 TOK_INT       int
   7 7.00008 282 TOK_IDENT     b
   7 7.00010  61 '='           =
   7 7.00012 283 TOK_INTCON    8
   7 7.00013  59 ';'           ;
   7 8.00004 262 TOK_INT       int
   7 8.00008 282 TOK_IDENT     c
   7 8.00010  61 '='           =
   7 8.00012 282 TOK_IDENT     a
   7 8.00014  43 '+'           +
   7 8.00016 282 TOK_IDENT     b
   7 8.00017  59 ';'           ;
   7 9.00004 282 TOK_IDENT     a
   7 9.00006  61 '='           =
   7 9.00008 282 TOK_IDENT     b
   7 9.00010  43 '+'           +
   7 9.00012 282 TOK_IDENT     c
   7 9.00013  59 ';'           ;
   7 10.00004 282 TOK_IDENT     putint
   7 10.00011  40 '('           (
   7 10.00012 282 TOK_IDENT     a
   7 10.00013  41 ')'           )
   7 10.00014  59 ';'           ;
   7 11.00004 282 TOK_IDENT     putchr
   7 11.00011  40 '('           (
   7 11.00012 284 TOK_CHARCON   '\n'
   7 11.00016  41 ')'           )
   7 11.00017  59 ';'           ;
   7 12.00004 268 TOK_RETURN    return
   7 12.00011 283 TOK_INTCON    0
   7 12.00012  59 ';'           ;
   7 13.00001 125 '}'           }
