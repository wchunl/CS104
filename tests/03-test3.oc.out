; 0.  1: # 1 "03-test3.oc"; 1.  0: 
; 1.  1: # 1 "<built-in>"
; 2.  1: # 1 "<command-line>"
; 3.  1: # 1 "03-test3.oc"
; 4.  1: 
; 4.  2: 
; 4.  3: # 1 "oclib.h" 1
; 5.  1: # 22 "oclib.h"
; 6. 22: void fail (string expr, string file, int line);
; 6. 23: 
; 6. 24: void putchr (int chr);
; 6. 25: void putint (int num);
; 6. 26: void putstr (string str);
; 6. 27: 
; 6. 28: int getchr();
; 6. 29: string getstr();
; 6. 30: string getln();
; 6. 31: # 4 "03-test3.oc" 2
; 7.  4: 
; 7.  5: int main() {
; 7.  6:    int a = 3;
; 7.  7:    int b = 8;
; 7.  8:    int c = a + b;
; 7.  9:    a = b + c;
; 7. 10:    putint (a);
; 7. 11:    putchr ('\n');
; 7. 12:    return 0;
; 7. 13: }
